frontend/packages/git-service/src/utils/pac-strategy-detector.ts

```diff
@@ -0,0 +1,54 @@
+import { BaseService } from '../services/base-service';
+
+export const detectPacFiles = async (gitService: BaseService) => {
+  let pacFiles = [];
+  const isTektonFolderPresent = await gitService.isTektonFolderPresent();
+
+  if (isTektonFolderPresent) {
+    pacFiles.push('.tekton');
+    const { files: pipelineFiles } = await gitService.getRepoFileList({
+      specificPath: '.tekton',
+      includeFolder: true,
+    });
+    if (pipelineFiles.length > 0) {
+      pacFiles = [...pacFiles, ...pipelineFiles];
+    }
+  }
+  return pacFiles;
+};
+
+export const verifyPipelinesFile = async (file: string, gitService: BaseService) => {
+  const trimmedFile = file?.startsWith('.tekton/') ? file?.replace('.tekton/', '') : file;
+
+  const resourceContent = await gitService.getFileContent(`.tekton/${trimmedFile}`);
+  return resourceContent.includes('kind: PipelineRun');
```

Comment:
`includes` is a really simple check. I don't say we need to parse the yaml here, but can we check that the file contains at least this string in a single line? So that there is no comment before this:


```yaml
kind: Pipeline
# TODO: kind: PipelineRun 
```

```suggestion
  return resourceContent.split(/\r?\n/).includes('kind: PipelineRun');
```

Reply:
Thanks. That's a very valid point